File: 240 Search a 2D Matrix II/solution.py
class Solution:
    # @param {integer[][]} matrix
    # @param {integer} target
    # @return {boolean}
    def searchMatrix(self, matrix, target):
        if not len(matrix):
            return False
        n,m = len(matrix)-1,len(matrix[0])
        l = 0
        while n>=0 and l<m:
            helper = matrix[n][l]
            if helper==target:
                return True
            elif helper<target:
                l+=1
            else:
                n-=1
        return False
        

File: 240 Search a 2D Matrix II/solution.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if (target < matrix[0][0]) return false;
        int row = matrix.size();
        int col = matrix[0].size();
        int cur_row = row - 1, cur_col = 0;
        while (cur_row >= 0 && cur_col < col) {
            if (matrix[cur_row][cur_col] == target)
                return true;
            else if (matrix[cur_row][cur_col] < target)
                cur_col++;
            else
                cur_row--;
        }
        return false;
    }
};



